doc/source/conf.py

```diff
@@ -70,19 +89,129 @@ def autodoc_skip_member(app, what, name, obj, skip, options):
+    """
+    # Get the full path to the document
+    docpath = os.path.join(app.srcdir, docname)
+
+    # Check if this is a .py example file
+    if not os.path.exists(docpath + '.py') or not docname.startswith("examples"):
+        return
+
+    logger.info(f"Adding '# %%time' to file {docname}.py")
+    lines = source[0].split("\n")
+    modified_lines = []
+    in_code_cell = False
+    in_code_cell_plus = False
+
+    for line in lines:
+        stripped_line = line.strip()
+        # Detect the start of a new code cell
+        if stripped_line.startswith('# +'):
+            in_code_cell = True
+            in_code_cell_plus = True
+            modified_lines.append(line)
+            modified_lines.append('# %%time')
+        # Detect the end of a code cell
+        elif stripped_line.startswith('# -'):
+            in_code_cell = False
+            in_code_cell_plus = False
+            modified_lines.append(line)
+        # Detect already being in a code cell
+        elif in_code_cell and in_code_cell_plus:
+            modified_lines.append(line)
+        elif in_code_cell and not in_code_cell_plus:
+            # Detect being out of a code cell
+            if stripped_line == "":
+                in_code_cell = False
+            modified_lines.append(line)
+        elif not in_code_cell:
+            # Detect the start of a new code cell
+            if not stripped_line.startswith("# ") and stripped_line not in ("", "#"):
+                in_code_cell = True
+                modified_lines.append('# %%time')
+                modified_lines.append(line)
+            # Detect already being out of a code cell
+            else:
+                modified_lines.append(line)
+        else:
+            raise Exception("not handled")
+
+    # Update the source
+    source[0] = "\n".join(modified_lines)
+    # logger.info(source[0])
+
+def adjust_image_path(app, docname, source):
+    """Adjust the HTML label used to insert images in the examples.
+    
+    The following path makes the examples in the root directory work:
+    # <img src="../../doc/source/_static/diff_via.png" width="500">
+    However, examples fail when used through the documentation build because
+    reaching the associated path should be "../../_static/diff_via.png".
+    Indeed, the directory ``_static`` is automatically copied into the output directory
+    ``_build/html/_static``.
+    """
+    # Get the full path to the document
+    docpath = os.path.join(app.srcdir, docname)
+
+    # Check if this is a PY example file
+    if not os.path.exists(docpath + '.py') or not docname.startswith("examples"):
+        return
+
+    logger.info(f"Changing HTML image path in '{docname}.py' file.")
+    source[0] = source[0].replace('../../doc/source/_static', '../../_static')
+
+def remove_ipython_time_from_html(app, pagename, templatename, context, doctree):
+    """Remove '# %%time' from examples generated HTML files.
```

Comment:
```suggestion
    """Remove '# %%time' from examples generated HTML files.
```
Missing word? ... from examples generated [to/in/from?] HTML files."